sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/timeExpressions.scala

```diff
@@ -162,6 +163,186 @@ object TryToTimeExpressionBuilder extends ExpressionBuilder {
   }
 }
 
+/**
+ * Converts or extracts time from various input types.
+ */
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = """
+    _FUNC_(expr) - Extracts the time part from or converts the given expression to a time.
+    The expression can be a string, timestamp, or numeric value.
+  """,
+  arguments = """
+    Arguments:
+      * expr - An expression that can be one of:
+          - A string representing a time
+          - A timestamp value
+          - A numeric value representing total seconds
+  """,
+  examples = """
+    Examples:
+      > SELECT _FUNC_('12:25:13.45');
+       12:25:13.45
+      > SELECT _FUNC_(timestamp'2020-04-30 12:25:13.45');
+       12:25:13.45
+      > SELECT _FUNC_(123);
+       00:02:03
+  """,
+  group = "datetime_funcs",
+  since = "4.1.0")
+// scalastyle:on line.size.limit
+case class Time(expr: Expression)
+  extends UnaryExpression with ExpectsInputTypes {
+
+  override def child: Expression = expr
+
+  override def dataType: DataType = TimeType()
+
+  override def inputTypes: Seq[AbstractDataType] = Seq(AnyDataType)
+
+  override def prettyName: String = "time"
+
+  override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+    val castExpr = Cast(child, TimeType())
+    castExpr.genCode(ctx)
+  }
+
+  override protected def nullSafeEval(input: Any): Any = {
+    Cast(child, TimeType()).eval(null)
```

Comment:
This is not right way of implementing a replacement. Please, take a look at
https://github.com/apache/spark/blob/203942c583187b6d0a012ff2b2d6aab5c664bd39/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala#L2180-L2184